compiler/plugins/target/AMD-AIE/iree-amd-aie/Transforms/Passes.cpp

```diff
@@ -788,10 +786,13 @@ void addMLIRAIRLoweringPasses(OpPassManager &passManager, AMDAIEDevice device) {
 }
 
 void addMLIRAIELoweringPasses(OpPassManager &passManager) {
-  passManager.addPass(createLowerAffinePass());
   OpPassManager &devicePM = passManager.nest<xilinx::AIE::DeviceOp>();
-  devicePM.addPass(createAMDAIEAssignLockIDsPass());
   devicePM.addPass(createAMDAIEObjectFifoStatefulTransformPass());
+  devicePM.addPass(createAMDAIEDmaToNpuPass());
+  passManager.addPass(createCanonicalizerPass());
+  passManager.addPass(createConvertLinalgToLoopsPass());
+  passManager.addPass(createLowerAffinePass());
+  devicePM.addPass(createAMDAIEAssignLockIDsPass());
```

Comment:
umm I don't think this does what it looks like it does; this creates a passpipeline like

`builtin.module(
  CanonicalizerPass,
  ConvertLinalgToLoopsPass,
  LowerAffinePass,
  device(
    AMDAIEObjectFifoStatefulTransformPass,
    AMDAIEDmaToNpuPass,
    AMDAIEAssignLockIDsPass,
    ...
  ),
  ConvertSCFToCFPass,
  device(AMDAIELocalizeLocksPass),
  device(AMDAIENormalizeAddressSpacesPass),
))
```

if you do `passMaanger.dump()` you'll see.

Reply:
Great spot, thanks